### PR TITLE
internal/gocdk: add support for provisioning runtimevar provider types

### LINKED ISSUE #2351: internal/gocdk: add support for provisioning runtimevars

### PR DESCRIPTION
**Pull request by vangent**
Adds support for `filevar`, `gcpruntimeconfig`, `awsparamstore`.

Fixes #2351.

### CODE REVIEW

File: internal/cmd/gocdk/internal/static/_assets/provision/runtimevar/gcpruntimeconfig.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# TODO(rvangent): Add comments explaining.
+
+locals {
+  gcpruntimeconfig_url  = "gcpruntimeconfig://projects/${local.gcp_project}/configs/${google_runtimeconfig_config.config.name}/variables/${google_runtimeconfig_variable.var.name}?decoder=string"
```
Comment (vangent) [inline internal/cmd/gocdk/internal/static/_assets/provision/runtimevar/gcpruntimeconfig.tf]: This should probably not be `local.gcp_project` here, but something referencing the `google` provider. I wasn't sure how to do that....

**Comment by vangent** [comment]
Ping?

File: internal/cmd/gocdk/internal/static/_assets/provision/runtimevar/gcpruntimeconfig.tf (hunk 1 above)
Comment (zombiezen) [reply internal/cmd/gocdk/internal/static/_assets/provision/runtimevar/gcpruntimeconfig.tf]: Using the [`google_project` data source](https://www.terraform.io/docs/providers/google/d/google_project.html):

```terraform
data "google_project" "project" {}

locals {
  gcpruntimeconfig_url = "gcpruntimeconfig://projects/${data.google_project.project.project_id}/configs/..."
}
```

Ideally there should be only one of these data sources per Terraform module.

Comment (vangent) [reply internal/cmd/gocdk/internal/static/_assets/provision/runtimevar/gcpruntimeconfig.tf]: Done.

**Pull request merged**